src/firewall/nft.rs

```diff
@@ -1083,7 +1091,12 @@ fn make_dns_dnat_rule(dns_ip: &IpAddr, dns_port: u16) -> schema::NfListObject {
                 flags: None,
             })),
         ],
-    )
+    );
+
+    // rule should be first so it is ordered before the normal contianer DNAT
+    rule.handle = Some(0);
```

Comment:
Is this still necessary since you're using INSERT instead of ADD above?

Reply:
Maybe not, I can test tomorrow. You might be right that insert defaults to the first position